packages/flutter/test/painting/text_painter_test.dart

```diff
@@ -145,17 +145,73 @@ void main() {
       caretOffset = painter.getOffsetForCaret(ui.TextPosition(offset: text.length), ui.Rect.zero);
       expect(caretOffset.dx, painter.width);
 
-      // Test with trailing full-width space
-      const String textWithFullWidthSpace = 'A\u{3000}';
-      checkCaretOffsetsLtr(textWithFullWidthSpace);
-      painter.text = const TextSpan(text: textWithFullWidthSpace);
+      text = 'A';
+      painter.text = TextSpan(text: text);
       painter.layout();
-      caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 0), ui.Rect.zero);
-      expect(caretOffset.dx, 0);
-      caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 1), ui.Rect.zero);
-      expect(caretOffset.dx, painter.width / 2);
-      caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: textWithFullWidthSpace.length), ui.Rect.zero);
-      expect(caretOffset.dx, painter.width);
+      final double textAWidth = painter.width;
+      void checkUnicodeZsCategory(String charset) {
+        text = 'A$charset';
+        checkCaretOffsetsLtr(text);
+        painter.text = TextSpan(text: text);
+        painter.layout();
+        caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 0), ui.Rect.zero);
```

Comment:
Can you also verify that each input character is indeed a space, by laying out the painter at `maxWidth = 14`, and checks if there's only one line, and the line width is exactly 14? If one of the characters is not considered a trailing space then there will be 2 lines (or lineWidth would be greater than 14 if the character is not a line breaker).

Reply:
I added test case at [9c37308](https://github.com/flutter/flutter/pull/152215/commits/9c373083b9a0c1979e3992348224b80c79308571).